src/vsphere_cpi/spec/unit/cloud/vsphere/resources/cluster_spec.rb

```diff
@@ -262,6 +375,116 @@ def generate_host_property(mob:, name:, connection_state:, maintenance_mode:, me
       end
     end
 
+    describe 'host groups and vm groups are defined in AZ' do
+      describe '#vm_group' do
+        let(:cluster_config) do
+          instance_double(
+            'VSphereCloud::ClusterConfig',
+            name: 'fake-cluster-name',
+            resource_pool: 'fake-resource-pool',
+            host_group: 'vcpi-hg1',
+          )
+        end
+        it 'returns host_group name suffixed with _vm_group' do
+          expect(cluster.vm_group).to eq(cluster.host_group+VSphereCloud::Resources::Cluster::CLUSTER_VM_GROUP_SUFFIX)
+        end
+      end
+      context 'when host group name is longer than 100 characters' do
+        let(:cluster_config) do
+          instance_double(
+            'VSphereCloud::ClusterConfig',
+            name: 'fake-cluster-name',
+            resource_pool: 'fake-resource-pool',
+            host_group: 'vcpi-hg1'*20,
+          )
+        end
+        it 'returns a random name' do
+          expect(cluster.vm_group).to_not eq(cluster.host_group+VSphereCloud::Resources::Cluster::CLUSTER_VM_GROUP_SUFFIX)
+        end
+      end
+      describe '#host_group' do
+        context 'when host group is not defined in config' do
+          it 'returns nil' do
+            expect(cluster.host_group).to be(nil)
+          end
+        end
+        context 'when host group is defined in config' do
+          let(:cluster_config) do
+            instance_double(
+              'VSphereCloud::ClusterConfig',
+              name: 'fake-cluster-name',
+              resource_pool: 'fake-resource-pool',
+              host_group: 'fake-host-group'
+            )
+          end
+          it 'returns name of the defined host Group)' do
+            expect(cluster.host_group).to eq('fake-host-group')
+          end
+        end
+      end
+      describe '#vm_host_affinity_rule_name' do
+        context 'when host group is not defined(nil)' do
+          it 'returns nil' do
+            expect(cluster.vm_host_affinity_rule_name).to be(nil)
+          end
+        end
+        context 'when host group is defined' do
+          let(:cluster_config) do
+            instance_double(
+              'VSphereCloud::ClusterConfig',
+              name: 'fake-cluster-name',
+              resource_pool: 'fake-resource-pool',
+              host_group: 'fake-host-group',
```

Comment:
you could just use let(:host_group) in all the tests instead of redefining cluster_config with host_group.

Reply:
I did a bit of refactor. It would eliminate some redundancy but not much. i will keep it same for now.